api/src/main/java/io/opentelemetry/api/trace/ImmutableSpanContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.trace;
+
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.extension.memoized.Memoized;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+@AutoValue
+abstract class ImmutableSpanContext implements SpanContext {
```

Comment:
Not sure why this class is needed, if you just move the "@AutoValue" annotation on the SpanContext you achieve the same thing, and just call directly the ctor of the AutoValue, am I missing something?

Reply:
As far as I know, `AutoValue` doesn't support interfaces still.